CRM/Contact/BAO/Query.php

```diff
@@ -5837,10 +5837,11 @@ public function filterRelatedContacts(&$from, &$where, &$having) {
       else {
         $from .= $qcache['from'];
       }
-      $where = $qcache['where'];
+      $where .= $qcache['where'];
```

Comment:
I kind of remember this code area from the earlier PR but I don't remember if there was ever a time when $where was passed in to the function and is blank or didn't have "WHERE" in it already. I wonder if it's worth adding a guard against that, e.g. if $where does not contain "WHERE", then `$where = "WHERE (1) " . $qcache['where']`. Or something like that - just an example.

Reply:
I have added a check whether $where is empty.